### PR TITLE
Scripts/VioletHold: Minor improvements

### PR DESCRIPTION
**Pull request by Jildor**
**Changes proposed:**

-  Now we can teleport inside with gossip if event is in progress
-  Improve HavePlayers() check logic
-  Remove a pointless comment (suggested by shauren)
-  Move CheckWipe function to execute every 3 seconds instead every map update (suggested by Riztazz)

**Target branch(es):** 3.3.5/master

- [X] 3.3.5
- [ ] master

**Issues addressed:** Ref #17880 (don't close it)

**Tests performed:** tested in-game

### CODE REVIEW

File: src/server/scripts/Northrend/VioletHold/violet_hold.cpp [hunk 1]
```diff
@@ -472,12 +472,13 @@ class npc_sinclari_vh : public CreatureScript
                             task.Repeat(Seconds(5));
                             break;
                         case 8:
-                            me->SetVisible(false);
+                            _instance->SetData(DATA_MAIN_EVENT_STATE, IN_PROGRESS);
+                            // [1] GUID: Full: 0xF1300077C202E6DD Type: Creature Entry: 30658 Low: 190173
```
Comment (Shauren) [inline src/server/scripts/Northrend/VioletHold/violet_hold.cpp]: well thats kind of a pointless comment, it brings no info to the script :D it doesn't even have the rest of packet data (most likely was part of UPDATE_OBJECT, in which case I would expect to see some field changes)

**Comment by jackpoz** [comment]
is this ready to be merged ? with last version tested ingame, etc ?

**Comment by Jildor** [comment]
Yep ;)

**Review by Keader — APPROVED** [review]
Looks good to me

**Review by ccrs — APPROVED** [review]

**Review by ghost — APPROVED** [review]
Looks good to me.

#sign-off

**Comment by jackpoz** [comment]
Thanks for the PR :)

**Pull request merged**